tests/sparsity/test_common.py

```diff
@@ -225,3 +229,66 @@ def test_polynomial_schedule_per_optimizer_step(self, magnitude_algo_mock, speci
             if last_ref_level is not None:
                 mock.assert_called_once_with(pytest.approx(last_ref_level))
             mock.reset_mock()
+
+def test_create_rb_algo_with_per_layer_scheduler_and_per_layer_loss():
+    config = get_empty_config()
+    config['compression'] = {'algorithm': 'rb_sparsity', "params": {"schedule": 'per_layer'}}
+    _, compression_ctrl = create_compressed_model_and_algo_for_test(MockModel(), config)
+
+    # pylint: disable=protected-access
+    assert isinstance(compression_ctrl._loss, SparseLossForPerLayerSparsity)
+
+def test_rb_sparsity_per_layer_scheduler__can_set_sparsity_level_for_module():
+    config = get_empty_config()
+    config['compression'] = {'algorithm': 'rb_sparsity', "params": {"schedule": 'per_layer'}}
+    _, compression_ctrl = create_compressed_model_and_algo_for_test(MockModel(), config)
+
+    # pylint: disable=protected-access
+    assert list(compression_ctrl._loss.per_layer_target.values())[0] == 1
+
+    compression_ctrl._scheduler.set_sparsity_level_for_module('MockModel/NNCFLinear[field]', 0.7)
+    assert list(compression_ctrl._loss.per_layer_target.values())[0] == pytest.approx(0.3)
+
+def test_example_of_using_PerLayerSparsityScheduler():
+    # This is an example of how the PerLayerSparsityScheduler would be used
+    #  as an interface to the automatic local selection level sparsity algorithm.
+
+    class PerLayerAutomaticControlSparsityLevelForExample():
+        def __init__(self, scheduler: PerLayerSparsityScheduler):
+            self.scheduler = scheduler
+            self.num_update = 0
+
+        def do_update_sparsity_levels(self):
+            for minfo in self.scheduler.algo.sparsified_module_info:
+                self.scheduler.set_sparsity_level_for_module(minfo.module_name, 0.1 + self.num_update * 0.05)
```

Comment:
I was thinking of a more explicit example, where `set_sparsity_level_for_module` would be called with an explicit, hard-coded model-specific string as an argument, such as `"TwoConvTestModel/NNCFConv2d[conv1]"` or something along these lines.